docs/early-access/devcontainers.md

```diff
@@ -0,0 +1,292 @@
+
+1. Selecting "Open in VS Code Desktop" from the Coder web interface
+2. Using the Coder CLI with the container flag:
+
+```console
+coder open vscode --container keen_dijkstra my-workspace
+```
+
+While optimized for VS Code, other IDEs with dev containers support may also
+work.
+
+### Port Forwarding
+
+During the early access phase, port forwarding is limited to ports defined via
+[`appPort`](https://containers.dev/implementors/json_reference/#image-specific)
+in your `devcontainer.json` file.
+
+For example, with this `devcontainer.json` configuration:
+
+```json
+{
+	"appPort": ["8080:8080", "4000:3000"]
+}
+```
+
+You can forward these ports to your local machine using:
+
+```console
+coder port-forward my-workspace --tcp 8080,4000
+```
+
+This forwards port 8080 (local) -> 8080 (agent) -> 8080 (dev container) and port
+4000 (local) -> 4000 (agent) -> 3000 (dev container).
+
+## Dev Container Features
+
+You can use standard dev container features in your `devcontainer.json` file.
+Coder also maintains a
+[repository of features](https://github.com/coder/devcontainer-features) to
+enhance your development experience.
+
+Currently available features include:
+
+- [code-server](https://github.com/coder/devcontainer-features/blob/main/src/code-server)
+
+To use the code-server feature, add the following to your `devcontainer.json`:
+
+```json
+{
+	"features": {
+		"ghcr.io/coder/devcontainer-features/code-server:1": {
+			"port": 13337
+		}
+	},
+	"appPort": [13337]
+}
+```
+
+> [!NOTE]
+>
+> Remember to include the port in the `appPort` section to ensure proper port
+> forwarding.
+
+## Limitations during Early Access
+
+During the early access phase, the dev containers integration has the following
+limitations:
+
+- Changes to the `devcontainer.json` file require manual container recreation
+- Automatic port forwarding only works for ports specified in `appPort`
+- SSH access requires using the `--container` flag
+- Some devcontainer features may not work as expected
+
+These limitations will be addressed in future updates as the feature matures.
+
+## Comparison with Envbuilder-based Dev Containers
+
+| Feature        | Dev Containers (Early Access)          | Envbuilder Dev Containers                    |
+|----------------|----------------------------------------|----------------------------------------------|
+| Implementation | Direct `@devcontainers/cli` and Docker | Coder's Envbuilder                           |
+| Target users   | Individual developers                  | Platform teams and administrators            |
+| Configuration  | Standard `devcontainer.json`           | Terraform templates with Envbuilder          |
+| Management     | User-controlled                        | Admin-controlled                             |
+| Requirements   | Docker access in workspace             | Compatible with more restricted environments |
+
+Choose the appropriate solution based on your team's needs and infrastructure
+constraints.
```

Comment:
Suggest linking to https://github.com/coder/envbuilder/blob/main/docs/devcontainer-spec-support.md for additional context.